### PR TITLE
support for passing template_kind hash as input parameter during create and update operations

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

File: app/controllers/api/v1/config_templates_controller.rb [hunk 1]
```diff
@@ -91,6 +92,11 @@ def default_template_url template, hostgroup
                 :id        => template.name, :hostgroup => hostgroup.name
       end
 
+      def process_template_kind
+        return unless params[:config_template] and (tk=params[:config_template].delete(:template_kind))
```
Comment (ohadlevy) [inline app/controllers/api/v1/config_templates_controller.rb:96]: maybe I'm missing something, but why do where do we pass template_kind and not template_kind_id ?

**Comment by dmitri-d** [comment]
when a config_template is serialized, template kind is serialized into a hash containing name and id attributes. Until this fix it wasn't possible to send the same serialization back.

**Comment by ohadlevy** [comment]
merged - thanks, ideally, next time create a bug at theforeman.org so it would be easy to track

**Comment by dmitri-d** [comment]
apologies, didn't think of it as a bug. will file a bug report next time.

**Pull request closed without merging**